### PR TITLE
Task vv/338 add spinner to mobile app

### PR DESCRIPTION
**Pull request by vitalik-prog**

### CODE REVIEW

File: src/screens/Podcast/styles.ts [hunk 1]
```diff
@@ -22,6 +22,13 @@ type Styles = {
 }
 
 const styles = StyleSheet.create<Styles>({
+  spinner: {
+    flexGrow: 1,
+    height: '100%',
+    flexDirection: 'column',
```
Comment (Denidzio) [inline src/screens/Podcast/styles.ts]:
```suggestion
```

File: src/screens/Podcast/styles.ts [hunk 2]
```diff
@@ -22,6 +22,13 @@ type Styles = {
 }
 
 const styles = StyleSheet.create<Styles>({
+  spinner: {
```
Comment (Denidzio) [inline src/screens/Podcast/styles.ts]: We will repeat same styles in every component. Let's put it to Sprinner/styles.ts

**Review by Denidzio — CHANGES_REQUESTED** [review]

File: src/screens/Podcast/styles.ts (hunk 2 above)
Comment (vitalik-prog) [reply src/screens/Podcast/styles.ts]: What if this spinner will be needed in some block? Not for all page. For example here src/screens/Podcast/components/EpisodesList/EpisodesList.tsx line 29 and line 34

File: src/screens/Podcast/styles.ts (hunk 1 above)
Comment (vitalik-prog) [reply src/screens/Podcast/styles.ts]: Sorry, but i don't understand it.

File: src/screens/Podcast/styles.ts (hunk 2 above)
Comment (eugeneprib) [reply src/screens/Podcast/styles.ts]: We should have default styles, and if we need we can't change it by passing the new styles down as props

File: src/components/Spinner/Spinner.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  wrapperStyle?: ViewStyle
+  type?: SpinnerType
+  size?: SpinnerSize
+}
+
+const Spinner: React.FC<Props> = ({
+  wrapperStyle,
+  type = DEFAULT_TYPE,
+  size = SpinnerSize.LARGE
+}) => {
```
Comment (eugeneprib) [inline src/components/Spinner/Spinner.tsx]:
```suggestion
const Spinner: React.FC<Props> = (props) => {
const {
  wrapperStyle,
  type = DEFAULT_TYPE,
  size = SpinnerSize.LARGE
 } = props;
```

**Review by eugeneprib — CHANGES_REQUESTED** [review]

File: src/screens/Podcast/styles.ts (hunk 2 above)
Comment (vitalik-prog) [reply src/screens/Podcast/styles.ts]: Thnx! It will be best variant

File: src/components/Spinner/common/constants/index.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import { SpinnerType } from '../enum'
+
+const DEFAULT_TYPE: SpinnerType = SpinnerType.PRIMARY
+const BASE_APP_COLOUR = '#1E1E1E'
```
Comment (eugeneprib) [inline src/components/Spinner/common/constants/index.ts]: What does this "APP_COLOUR" mean?

**Review by eugeneprib — APPROVED** [review]

File: src/screens/Podcast/styles.ts (hunk 1 above)
Comment (Denidzio) [reply src/screens/Podcast/styles.ts]: Oh, I'm sorry, I mean delete this.

File: src/components/Spinner/common/constants/index.ts (hunk 4 above)
Comment (vitalik-prog) [reply src/components/Spinner/common/constants/index.ts]: This is our main application colour according to design. I thought it good name.

Comment (eugeneprib) [reply src/components/Spinner/common/constants/index.ts]: It's a good name, but it should be in a global config, not here

File: src/components/Spinner/Spinner.tsx (hunk 3 above)
Comment (Denidzio) [reply src/components/Spinner/Spinner.tsx]: > 

Why did we do that?

File: src/components/Spinner/common/constants/index.ts (hunk 4 above)
Comment (vitalik-prog) [reply src/components/Spinner/common/constants/index.ts]: Yes i take it out to global constants and change everywhere. We should do it at start project

File: src/components/Spinner/Spinner.tsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+const Spinner: React.FC<Props> = (props) => {
+  const {
+    wrapperStyle = styles.defaultWrapper,
+    type = DEFAULT_TYPE,
+    size = SpinnerSize.LARGE
+  } = props
+
+  const typeStyles = getStylesByType(type)
```
Comment (Denidzio) [inline src/components/Spinner/Spinner.tsx]:
```suggestion
```

File: src/common/constants/colours.ts [hunk 6]
```diff
@@ -0,0 +1,4 @@
+const BASE_APP_COLOUR = '#1E1E1E'
```
Comment (Denidzio) [inline src/common/constants/colours.ts]: Let's do it as Enum. For example `AppColour`.

File: src/components/Button/common/helpers/getStylesByType.ts [hunk 7]
```diff
@@ -6,7 +7,7 @@ const getStylesByType = (type: ButtonType): Styles => {
     case ButtonType.PRIMARY: {
       return {
         touchable: {
-          backgroundColor: '#1E1E1E'
+          backgroundColor: BASE_APP_COLOUR
```
Comment (Denidzio) [inline src/components/Button/common/helpers/getStylesByType.ts]:
```suggestion
          backgroundColor: AppColour.PRIMARY
```

File: src/components/Button/common/helpers/getStylesByType.ts [hunk 8]
```diff
@@ -16,7 +17,7 @@ const getStylesByType = (type: ButtonType): Styles => {
     case ButtonType.SECONDARY: {
       return {
         touchable: {
-          borderColor: '#f3427f',
+          borderColor: SECONDARY_APP_COLOUR,
```
Comment (Denidzio) [inline src/components/Button/common/helpers/getStylesByType.ts]:
```suggestion
          borderColor: AppColour.SECONDARY,
```

File: src/components/Spinner/Spinner.tsx [hunk 9]
```diff
@@ -0,0 +1,30 @@
+import React from 'react'
+import { View, ActivityIndicator, ViewStyle } from 'react-native'
+import { DEFAULT_TYPE } from './common/constants'
+import { SpinnerType, SpinnerSize } from './common/enum'
+import { getStylesByType } from './common/helpers'
+import styles from './styles'
+
+type Props = {
+  wrapperStyle?: ViewStyle
+  type?: SpinnerType
```
Comment (Denidzio) [inline src/components/Spinner/Spinner.tsx]: color
```suggestion
  colour?: AppColour
```

File: src/components/Spinner/Spinner.tsx [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+  size?: SpinnerSize
+}
+
+const Spinner: React.FC<Props> = (props) => {
+  const {
+    wrapperStyle = styles.defaultWrapper,
+    type = DEFAULT_TYPE,
```
Comment (Denidzio) [inline src/components/Spinner/Spinner.tsx]:
```suggestion
    colour = AppColour.PRIMARY,
```

File: src/components/Spinner/Spinner.tsx [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+    size = SpinnerSize.LARGE
+  } = props
+
+  const typeStyles = getStylesByType(type)
+
+  return (
+    <View style={wrapperStyle}>
+      <ActivityIndicator size={size} color={typeStyles} />
```
Comment (Denidzio) [inline src/components/Spinner/Spinner.tsx]:
```suggestion
      <ActivityIndicator size={size} color={colour} />
```

File: src/components/Spinner/common/constants/index.ts [hunk 12]
```diff
@@ -0,0 +1,5 @@
+import { SpinnerType } from '../enum'
```
Comment (Denidzio) [inline src/components/Spinner/common/constants/index.ts:1]: Delete it

File: src/components/Spinner/common/enum/spinnerType.ts [hunk 13]
```diff
@@ -0,0 +1,6 @@
+enum SpinnerType {
```
Comment (Denidzio) [inline src/components/Spinner/common/enum/spinnerType.ts]: Delete it

File: src/components/Spinner/common/helpers/getStylesByType.ts [hunk 14]
```diff
@@ -0,0 +1,15 @@
+import { BASE_APP_COLOUR, SECONDARY_APP_COLOUR } from 'src/common/constants'
```
Comment (Denidzio) [inline src/components/Spinner/common/helpers/getStylesByType.ts]: It's not styles, it's just colour. It's redundant. Remove, please

**Review by Denidzio — CHANGES_REQUESTED** [review]
I recommend you use `color` instead of `colour`

File: src/screens/Podcast/Podcast.tsx [hunk 15]
```diff
@@ -24,26 +18,34 @@ import {
...
+    dataStatus: podcast.dataStatus,
+    totalCount: podcast.totalCount,
+    hasMoreEpisodes: podcast.hasMoreEpisodes,
+    episodesDataStatus: podcast.episodesDataStatus
+  }))
 
   const route = useRoute<PodcastScreenRouteProp>()
   const navigation = useNavigation<PodcastScreenNavigationProp>()
 
   const isLoading = dataStatus === DataStatus.PENDING
```
Comment (Denidzio) [inline src/screens/Podcast/Podcast.tsx]:
```suggestion
  const isPodcastFetching = dataStatus === DataStatus.PENDING
```

File: src/screens/Podcast/Podcast.tsx [hunk 16]
```diff
@@ -73,18 +75,14 @@ const Podcast: React.FC = () => {
...
-        <ActivityIndicator size="large" color="#f3427f" />
-      </View>
-    )
-  }
-
   const handleNavigateToHome = () => {
     navigation.navigate(NavigationScreen.HOME)
   }
 
+  if (isLoading) {
```
Comment (Denidzio) [inline src/screens/Podcast/Podcast.tsx]:
```suggestion
  if (isPodcastFetching) {
```

File: src/components/Spinner/index.tsx [hunk 17]
```diff
@@ -0,0 +1 @@
+export { default } from './Spinner'
```
Comment (Denidzio) [inline src/components/Spinner/index.tsx:1]:
```suggestion
export { default } from './Spinner'
export { SpinnerSize } from './common/enums' 
```

File: src/components/index.ts [hunk 18]
```diff
@@ -3,3 +3,4 @@ export { default as Input } from './Input'
 export { default as Link } from './Link'
 export { default as Heading, HeadingType } from './Heading'
 export { default as PlainText } from './PlainText'
+export { default as Spinner } from './Spinner'
```
Comment (Denidzio) [inline src/components/index.ts]:
```suggestion
export { default as Spinner, SpinnerSize } from './Spinner'
```

**Review by Denidzio — CHANGES_REQUESTED** [review]

**Review by Denidzio — APPROVED** [review]
Looks good to me 👍🏻

File: src/components/Spinner/Spinner.tsx (hunk 3 above)
Comment (eugeneprib) [reply src/components/Spinner/Spinner.tsx]: To make it more readable

File: src/components/Spinner/Spinner.tsx (hunk 5 above)
Comment (eugeneprib) [reply src/components/Spinner/Spinner.tsx]: It's ok both

**Pull request merged**